pkg/chunk/cassandra/storage_client.go

```diff
@@ -117,9 +121,18 @@ func (cfg *Config) setClusterConfig(cluster *gocql.ClusterConfig) error {
 	cluster.DisableInitialHostLookup = cfg.DisableInitialHostLookup
 
 	if cfg.SSL {
-		cluster.SslOpts = &gocql.SslOptions{
-			CaPath:                 cfg.CAPath,
-			EnableHostVerification: cfg.HostVerification,
+		if cfg.HostVerification {
+			cluster.SslOpts = &gocql.SslOptions{
+				CaPath:                 cfg.CAPath,
+				EnableHostVerification: true,
+				Config: &tls.Config{
+					ServerName: strings.Split(cfg.Addresses, ",")[0],
+				},
```

Comment:
This part is not required. When `ServerName` is set to "", then it will be inferred from hostname we're connecting to (in `tls.DialWithDialer` function). This also removes restriction that only single address can be used.

Reply:
ServerName must be set.  Gocql doesnt use tls.DialWithDialer.  It creates a TCP connection to the cassandra host, then if TLS is configured it upgrades the connection via tls.Client(conn, cfg.tslConfig).

the [crypto/tls docs](https://golang.org/pkg/crypto/tls/#Client) are clear that when using tls.Client(), "users must set either ServerName or InsecureSkipVerify in the config"

Reply:
@woodsaj Thanks for explanation!